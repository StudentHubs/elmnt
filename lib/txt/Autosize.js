"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var refluent_1 = require("refluent");
var utils_1 = require("../utils");
exports.default = refluent_1.default
    .do(utils_1.restyle('rows', function (rows, style) {
    return style.merge({
        visibility: 'hidden',
        whiteSpace: 'pre-wrap',
        wordBreak: 'break-word',
        minHeight: parseFloat(style.lineHeight) * (rows || 1),
        display: 'block',
        overflow: 'hidden',
    });
}))
    .yield(function (_a) {
    var value = _a.value, rows = _a.rows, style = _a.style;
    return (React.createElement("span", { style: style },
        value || '',
        rows ? '\n' : '',
        "."));
});
