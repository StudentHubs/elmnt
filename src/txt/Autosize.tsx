import * as React from 'react';
import { branch, compose, mapProps, renderNothing, setDisplayName } from 'recompose';
import css, { mapStyle } from 'highstyle';

export interface AutosizeStyle extends React.CSSProperties {
  lineHeight: string;
}
export interface AutosizeProps {
  value: string;
  rows?: number;
  style: AutosizeStyle;
}
export default compose<any, AutosizeProps>(

  setDisplayName('Autosize'),

  branch(({ rows }) => !rows, renderNothing),

  mapStyle(({ rows, style }) => [
    css.merge({
      visibility: 'hidden',
      whiteSpace: 'pre-wrap',
      wordBreak: 'break-word',
      minHeight: parseFloat(style.lineHeight) * rows,
      display: 'block',
    }),
  ]),

  mapProps(({ value, style }) => ({
    style,
    children: value.replace(/\n$/, '\n.'),
  })),

)('span' as any);
