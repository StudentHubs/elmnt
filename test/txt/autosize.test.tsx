import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Autosize from '../../src/txt/Autosize';

describe('txt: autosize', () => {

  it('single line of text, no rows', () => {
    expect(renderer.create(
      <Autosize value="Hello!" style={{ lineHeight: '30px', width: 100 }} />
    ).toJSON()).toMatchSnapshot();
  });

  it('single line of text, single row', () => {
    expect(renderer.create(
      <Autosize value="Hello!" rows={1} style={{ lineHeight: '30px', width: 100 }} />
    ).toJSON()).toMatchSnapshot();
  });

  it('single line of text, single row, overwritten style', () => {
    expect(renderer.create(
      <Autosize value="Hello!" rows={1} style={{ lineHeight: '30px', visibility: 'visible' }} />
    ).toJSON()).toMatchSnapshot();
  });

  it('single line of text, three rows', () => {
    expect(renderer.create(
      <Autosize value="Hello!" rows={3} style={{ lineHeight: '30px', width: 100 }} />
    ).toJSON()).toMatchSnapshot();
  });

  it('multiple lines of text, no rows', () => {
    expect(renderer.create(
      <Autosize value={'Hello\nWorld!'} style={{ lineHeight: '30px', width: 100 }} />
    ).toJSON()).toMatchSnapshot();
  });

  it('multiple lines of text, single row', () => {
    expect(renderer.create(
      <Autosize value={'Hello\nWorld!'} rows={1} style={{ lineHeight: '30px', width: 100 }} />
    ).toJSON()).toMatchSnapshot();
  });

  it('multiple lines of text, three rows', () => {
    expect(renderer.create(
      <Autosize value={'Hello\nWorld!'} rows={3} style={{ lineHeight: '30px', width: 100 }} />
    ).toJSON()).toMatchSnapshot();
  });

});
