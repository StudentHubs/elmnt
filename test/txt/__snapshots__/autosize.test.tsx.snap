// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`txt: autosize multiple lines of text, no rows 1`] = `null`;

exports[`txt: autosize multiple lines of text, single row 1`] = `
<span
  style={
    Object {
      "display": "block",
      "lineHeight": "30px",
      "minHeight": 30,
      "visibility": "hidden",
      "whiteSpace": "pre-wrap",
      "width": 100,
      "wordBreak": "break-word",
    }
  }
>
  Hello
  World!
</span>
`;

exports[`txt: autosize multiple lines of text, three rows 1`] = `
<span
  style={
    Object {
      "display": "block",
      "lineHeight": "30px",
      "minHeight": 90,
      "visibility": "hidden",
      "whiteSpace": "pre-wrap",
      "width": 100,
      "wordBreak": "break-word",
    }
  }
>
  Hello
  World!
</span>
`;

exports[`txt: autosize single line of text, no rows 1`] = `null`;

exports[`txt: autosize single line of text, single row 1`] = `
<span
  style={
    Object {
      "display": "block",
      "lineHeight": "30px",
      "minHeight": 30,
      "visibility": "hidden",
      "whiteSpace": "pre-wrap",
      "width": 100,
      "wordBreak": "break-word",
    }
  }
>
  Hello!
</span>
`;

exports[`txt: autosize single line of text, single row, overwritten style 1`] = `
<span
  style={
    Object {
      "display": "block",
      "lineHeight": "30px",
      "minHeight": 30,
      "visibility": "hidden",
      "whiteSpace": "pre-wrap",
      "wordBreak": "break-word",
    }
  }
>
  Hello!
</span>
`;

exports[`txt: autosize single line of text, three rows 1`] = `
<span
  style={
    Object {
      "display": "block",
      "lineHeight": "30px",
      "minHeight": 90,
      "visibility": "hidden",
      "whiteSpace": "pre-wrap",
      "width": 100,
      "wordBreak": "break-word",
    }
  }
>
  Hello!
</span>
`;
